src/Craftify.Geometry.Domain/LineSegmentGraphExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Craftify.Geometry.Domain;
+
+public static class LineSegmentGraphExtensions
+{
+    public static IReadOnlyList<Point3D> DistinctedNodes(this LineSegmentsGraph graph)
+    {
+        return ReadOnlyList.Of(graph.Edges
+            .SelectMany(edge => new[] { edge.Start, edge.End })
```

Comment:
I think we have a function to get end points from a Line,
```csharp
    public static IEnumerable<Point3D> GetEndPoints(this LineSegment lineSegment)
    {
        yield return lineSegment.Start;
        yield return lineSegment.End;
    }
```
```suggestion
            .SelectMany(edge => edge.GetEndPoints())
```